examples/bzlmod/MODULE.bazel

```diff
@@ -1,36 +1,122 @@
+# This file replaces the WORKSPACE file when using bzlmod.
+
+# module declares certain properties of the Bazel module represented by the current Bazel repo.
+# These properties are either essential metadata of the module (such as the name and version),
+# or affect behavior of the current module and its dependents.
 module(
     name = "example_bzlmod",
     version = "0.0.0",
     compatibility_level = 1,
 )
 
+# The following stanza defines the dependency rules_python.
+# For typical setups you set the version.
+# See the releases page for available versions.
+# https://github.com/bazelbuild/rules_python/releases
 bazel_dep(name = "rules_python", version = "0.0.0")
+
+# The following loads rules_python from the file system.
+# For usual setups you should remove this local_path_override block.
 local_path_override(
     module_name = "rules_python",
     path = "../..",
 )
 
+# The following stanza defines the dependency rules_python_gazelle_plugin.
+# For typical setups you set the version.
+# See the releases page for available versions.
+# https://github.com/bazelbuild/rules_python/releases
+bazel_dep(name = "rules_python_gazelle_plugin", version = "0.0.0")
+
+# The following starlark loads the gazelle plugin from the file system.
+# For usual setups you should remove this local_path_override block.
+local_path_override(
+    module_name = "rules_python_gazelle_plugin",
+    path = "../../gazelle",
+)
+
+# The following stanza defines the dependency for gazelle
+# See here https://github.com/bazelbuild/bazel-gazelle/releases/ for the
+# latest version.
+bazel_dep(name = "gazelle", version = "0.30.0", repo_name = "bazel_gazelle")
+
+# The following stanze returns a proxy object representing a module extension;
+# its methods can be invoked to create module extension tags.
 python = use_extension("@rules_python//python:extensions.bzl", "python")
+
+# This version is used in various places to load the correct
+# version of Python.
+PYTHON_VERSION = "3.9"
+
+# We next initaze the python toolchain using the extension.
+# You can set different Python versions in this block.
 python.toolchain(
-    name = "python3_9",
+    # This name is used in the various use_repo statements
+    # below, and in the local extension that is in this
+    # example.
+    name = "python3",
     configure_coverage_tool = True,
-    python_version = "3.9",
+    python_version = PYTHON_VERSION,
 )
-use_repo(python, "python3_9")
-use_repo(python, "python3_9_toolchains")
 
+# Import the python repositories generated by the given module extension into the scope of the current module.
+use_repo(python, "python3_toolchains")
+use_repo(python, "python3")
+
+# In bazel 6.2 you will be able to load these dynamically.
+# We need each of these repos load in order for setting the pyton_interpreter_target.
+# You only need the python repo for the operating systems that you are supporting.
+use_repo(python, "python3_x86_64-unknown-linux-gnu")
```

Comment:
I think I understand why these use_repo() calls are necessary, but can we hide this from users somehow? Because its only necessary for the pip setup, can we fold it into that extension somehow?

Reply:
I don't think so.